testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -237,6 +239,8 @@ protected void insertTableRows(String tableName, String floatToDoubleType)
                         "  '1234.01234', " +
                         "  'Infinity'," +
                         "  'NaN'," +
+                        "  DATE '2023-09-28', " +
+                        "  DATE '2000-04-13', " +
```

Comment:
nit: Can we set them to year like `10000` so that we could know how hive behaves for those date format.

Reply:
I've been trying to setup a test using a year > 10000 and while doing

```
ptl env up  --environment multinode
```

i noticed that the the `SELECT` from the Hive table before the coercion is failing with the below error:

```
java.lang.IllegalArgumentException
	at java.sql.Date.valueOf(Date.java:143)
	at org.apache.hive.jdbc.HiveBaseResultSet.evaluate(HiveBaseResultSet.java:451)
	at org.apache.hive.jdbc.HiveBaseResultSet.getColumnValue(HiveBaseResultSet.java:425)
	at org.apache.hive.jdbc.HiveBaseResultSet.getObject(HiveBaseResultSet.java:467)
	at org.apache.hive.beeline.Rows$Row.<init>(Rows.java:160)
	at org.apache.hive.beeline.BufferedRows.<init>(BufferedRows.java:57)
	at org.apache.hive.beeline.IncrementalRowsWithNormalization.<init>(IncrementalRowsWithNormalization.java:50)
	at org.apache.hive.beeline.BeeLine.print(BeeLine.java:2305)
```

We can change the type of the column to `string` and select then something like `000-01-04` (for an initial value of `10000-01-04`), but i'm not sure whether this makes any sense.

cc @dain

Reply:
```
0: jdbc:hive2://localhost:10000/default> select datediff(d, date '1970-01-01') from t3;
INFO  : Compiling command(queryId=hive_20231204192218_64b185e9-3181-42c0-b7a0-39151995cc6d): select datediff(d, date '1970-01-01') from t3
INFO  : Semantic Analysis Completed (retrial = false)
INFO  : Returning Hive schema: Schema(fieldSchemas:[FieldSchema(name:_c0, type:int, comment:null)], properties:null)
INFO  : Completed compiling command(queryId=hive_20231204192218_64b185e9-3181-42c0-b7a0-39151995cc6d); Time taken: 0.134 seconds
INFO  : Executing command(queryId=hive_20231204192218_64b185e9-3181-42c0-b7a0-39151995cc6d): select datediff(d, date '1970-01-01') from t3
INFO  : Completed executing command(queryId=hive_20231204192218_64b185e9-3181-42c0-b7a0-39151995cc6d); Time taken: 0.0 seconds
INFO  : OK
+----------+
|   _c0    |
+----------+
| 2932900  |
+----------+
```

Reply:
I've spent some extra time trying to shape a way to bring the requested scenario in the product test. However, I'm thinking that the value itself is not very representative because Hive behaves erroneously for decoding the value specifically for ORC storage format.
I've tried 

```
select datediff(date(d), date('1970-01-01')) from orctable
```
and received the wrong result `-719525`.